kubecost/templates/kubecost-network-costs-cluster-role-binding-template.yaml

```diff
@@ -0,0 +1,16 @@
+{{- if (.Values.networkCosts).enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "kubecost.serviceAccountName" . }}-network-costs-cluster-role
+  labels:
+    {{- include "kubecost.commonLabels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ template "kubecost.serviceAccountName" . }}-network-costs-cluster-role
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "kubecost.serviceAccountName" . }}
+    namespace: {{ .Release.Namespace }}
```

Comment:
This ClusterRoleBinding ties the Role to the `"kubecost.serviceAccountName"`. This service account has a number of other usages:
- in the frontend Deployment
- in the aggregator StatefulSet, in the absence of an aggregator-specific SA
- in the cloud cost Deployment, in the absence of a cloud cost-specific SA

We may want to consider the idea of setting up a separate, network costs daemonset-specific SA? In the worst case scenario, where network costs are enabled and this service account ends up being used for the other workloads, we would end up granting the ClusterRole permissions to all these workloads. I'm not sure how big of a concern this is?

Reply:
Yes, this should have its own SA as per Kubernetes best practices.

Reply:
Thanks for digging into `"kubecost.serviceAccountName"`! I agree that the network costs daemonset should have its own SA.

Reply:
I think Jesse would agree too, because he started a draft PR to separate out all ServiceAccounts https://github.com/kubecost/kubecost/pull/4216